src/azure-cli/azure/cli/command_modules/acs/custom.py

```diff
@@ -1833,17 +1862,23 @@ def aks_disable_addons(cmd, client, resource_group_name, name, addons, no_wait=F
     # send the managed cluster representation to update the addon profiles
     return sdk_no_wait(no_wait, client.create_or_update, resource_group_name, name, instance)
 
-
 def aks_enable_addons(cmd, client, resource_group_name, name, addons, workspace_resource_id=None,
                       subnet_name=None, no_wait=False):
     instance = client.get(resource_group_name, name)
     subscription_id = get_subscription_id(cmd.cli_ctx)
-    service_principal_client_id = instance.service_principal_profile.client_id
+
     instance = _update_addons(cmd, instance, subscription_id, resource_group_name, addons, enable=True,
                               workspace_resource_id=workspace_resource_id, subnet_name=subnet_name, no_wait=no_wait)
 
     if 'omsagent' in instance.addon_profiles:
         _ensure_container_insights_for_monitoring(cmd, instance.addon_profiles['omsagent'])
+        
+    # send the managed cluster representation to update the addon profiles
+    result = sdk_no_wait(no_wait, client.create_or_update,
+                          resource_group_name, name, instance)
+    result = LongRunningOperation(cmd.cli_ctx)(result)
```

Comment:
similarly here, I think, we should execute LongRunningOperation function only if the monitoring enabled. This way we dont have impact on other scenarios.